code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -143,22 +146,34 @@
 		icon_state = "[base_state][active]"
 
 /obj/item/weapon/melee/energy/sword/attackby(obj/item/weapon/W, mob/living/user)
+	if(combine(W, user))
+		return
 	..()
-	if(istype(W, /obj/item/weapon/melee/energy/sword))
+
+/obj/item/weapon/melee/energy/sword/proc/combine(obj/item/W, mob/living/user)
+	if(can_combine_with(W))
 		to_chat(user, "<span class='notice'>You attach the ends of the two energy swords, making a single double-bladed weapon! You're cool.</span>")
-		var/obj/item/weapon/dualsaber/saber = new /obj/item/weapon/dualsaber(user.loc)
+		var/obj/item/weapon/melee/energy/sword/dualsaber/saber = new dualsaber_type(user.loc)
 		saber.colorset = W._color + src._color
-		qdel(W)
-		W = null
-		qdel(src)
+		saber.swords.Add(W, src)
+		user.drop_item(W)
+		W.forceMove(saber)
+		user.drop_item(src)
+		forceMove(saber)
+		user.put_in_hands(saber)
+		return 1
+	return 0
 
+/obj/item/weapon/melee/energy/sword/proc/can_combine_with(obj/item/W)
+	return istype(W, /obj/item/weapon/melee/energy/sword) && !is_type_in_list(W, list(/obj/item/weapon/melee/energy/sword/bsword, /obj/item/weapon/melee/energy/sword/pirate, /obj/item/weapon/melee/energy/sword/dualsaber))
```

Comment:
Is it possible to replace these with a var on the e-swords where if the var is equal to the other item's var you can combine them?